### PR TITLE
Paper - Katrina Li

### PR DESCRIPTION
**Pull request by likatrina7**

### CODE REVIEW

File: tests/test_wave_07.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    item_e = Clothing(condition=3.5, age=6.0)
+
+    kate = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    result = kate.get_min_age()
+
+    assert result.category == "Electronics"
+    assert result.age == pytest.approx(0.0)
```
Comment (CheezItMan) [inline tests/test_wave_07.py:22]: Nice use of `pytest.approx`

File: tests/test_wave_07.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+def test_get_min_age_return_none():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:24]: Good check of an edge-case.  I also suggest testing what happens if two products in inventory have the same age.

File: tests/test_wave_07.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    assert result is True
+    assert len(mai.inventory) is 3
+    assert len(jesse.inventory) is 3
+    assert item_c not in jesse.inventory
+    assert item_a in jesse.inventory
+    assert item_e not in mai.inventory
+    assert item_f in mai.inventory
+
+def test_swap_by_newest_my_list_is_empty_return_false():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:60]: Nice edge-case tests when each vendors lists are empty.

I would also test and see what happens when each vendor has items of equal age.

File: swap_meet/clothing.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# wave 05
```
Comment (CheezItMan) [inline swap_meet/clothing.py:1]: 👍 Very clean child classes!

File: swap_meet/item.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:1]: 👍

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
+import sys
+
+class Vendor:
+    # wave 01
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory == None else inventory
```
Comment (CheezItMan) [inline swap_meet/vendor.py:6]: 👍 I'm glad you handled the default value for `inventory`

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+    def add(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:8]: 👍

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:12]: 👍

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        for i, thing in enumerate(self.inventory):
+            if thing == item:
+                return self.inventory.pop(i)
+
+        return False
+
+    # wave 02
+    def get_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:20]: 👍

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+
+        return items
+    
+    #  wave 03
+    def swap_items(self, friend, my_item, their_item):
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
```
Comment (CheezItMan) [inline swap_meet/vendor.py:31]: This could be a little simpler.  This works because `[]` is falsey.

```suggestion
        if self.inventory or friend.inventory:
```

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+            friend.add(my_item)
+            self.remove(my_item)
+            self.add(their_item)
+            friend.remove(their_item)
+            return True  
+
+        return False
+
+    #  wave 04
+    def swap_first_item(self, friend):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:44]: 👍 With a similar suggestion for simplicity.

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+            
+        return self.swap_items(friend, self.inventory[0], friend.inventory[0])    
+
+    #  wave 06
+    def get_best_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:51]: 👍

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        for item in self.inventory:
+            if item.category == category and \
+            item.condition >= highest_condition_rate:
+                highest_condition_rate = item.condition
+                best_item_by_category = item
+
+        return best_item_by_category
+
+    # wave 06
+    def swap_best_by_category(self, other, my_priority, their_priority):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:64]: 👍

File: swap_meet/vendor.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        their_best_offer = other.get_best_by_category(my_priority)
+        my_best_offer = self.get_best_by_category(their_priority)
+
+        swap_result = self.swap_items(other, my_best_offer, their_best_offer)
+
+        return swap_result
+
+    # optional wave 07
+    def get_min_age(self):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:73]: 👍 Love the docstrings btw.

File: swap_meet/vendor.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+        min_age = sys.maxsize
+
+        for item in self.inventory:
+            if item.age <= min_age:
+                newest_item = item
+                min_age = item.age
+        
+        return newest_item
+
+    def swap_by_newest(self, other):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:89]: 👍

**Review by CheezItMan — COMMENTED** [review]
Katrina is this is an outstanding submission.  You hit all the learning goals and went further with the optionals and tests for the optionals.  I like how you tested for edge cases and reused methods in `Vendor`.  Very well done.  

I left a few comments, mostly on style and testing edge-cases.
